controllers/imageupdateautomation_controller.go

```diff
@@ -255,11 +256,16 @@ func (r *ImageUpdateAutomationReconciler) Reconcile(ctx context.Context, req ctr
 		return failWithError(err)
 	}
 
-	lgc, err := libgit2pkg.NewClient(tmp, authOpts)
+	var gitClient git.RepositoryClient
+	if origin.Spec.GitImplementation == libgit2.ClientName {
```

Comment:
Switch statement would seem more appropriate here with a `default` which errors out for an unsupported implementation string?